wandb/apis/workspaces/interface.py

```diff
@@ -0,0 +1,449 @@
+import os
+from typing import Dict, Iterable, Literal, Optional
+from typing import List as LList
+from urllib.parse import parse_qs, urlparse, urlunparse
+
+from annotated_types import Annotated, Ge
+from pydantic import AnyUrl, ConfigDict, Field, PositiveInt
+from pydantic.dataclasses import dataclass
+
+import wandb
+from wandb.apis.reports.v2.interface import PanelTypes, _lookup_panel
+
+from . import expr, internal
+
+dataclass_config = ConfigDict(validate_assignment=True, extra="forbid", slots=True)
+
+
+def is_not_all_none(v):
+    if v is None or v == "":
+        return False
+    if isinstance(v, Iterable) and not isinstance(v, str):
+        return any(v not in (None, "") for v in v)
+    return True
+
+
+def is_not_internal(k):
+    return not k.startswith("_")
+
+
+def should_show(v):
+    """This is a workaround because BaseMetric.__eq__ returns FilterExpr."""
+    if isinstance(v, Iterable) and not isinstance(v, str):
+        return any(should_show(x) for x in v)
+    if isinstance(v, expr.BaseMetric):
+        return True
+    return False
+
+
+@dataclass(config=dataclass_config, repr=False)
+class Base:
+    def __repr__(self):
+        fields = (
+            f"{k}={v!r}"
+            for k, v in self.__dict__.items()
+            if (is_not_all_none(v) and is_not_internal(k)) or (should_show(v))
+        )
+        fields_str = ", ".join(fields)
+        return f"{self.__class__.__name__}({fields_str})"
+
+    def __rich_repr__(self):
+        for k, v in self.__dict__.items():
+            if (is_not_all_none(v) and is_not_internal(k)) or (should_show(v)):
+                yield k, v
+
+    @property
+    def _model(self):
+        return self.to_model()
+
+    @property
+    def _spec(self):
+        return self._model.model_dump(by_alias=True, exclude_none=True)
+
+
+@dataclass
+class SectionPanelSettings(Base):
```

Comment:
Building off on the comment on the `Section`:
I would then imagine this class to contain the x-axis and smoothing specific settings

```
@dataclass
class SectionSettings(Base):

    # these are the only line plot section level settings users can defined in UI right now
    # TODO - make this into it's own class?
    x_axis: str = "_step"  # fix this to use name map in future
    x_min: Optional[Annotated[float, Ge(0)]] = None
    x_max: Optional[Annotated[float, Ge(0)]] = None
    
    smoothing_type: internal.SmoothingType = "none"
    smoothing_weight: Annotated[float, Ge(0)] = 0

```